test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/HttpRequestMessageBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace Microsoft.Health.Dicom.Web.Tests.E2E.Rest
+{
+    public class HttpRequestMessageBuilder
```

Comment:
>HttpRequestMessageBuilder [](start = 17, length = 25)

This now creates two paths to create requests - this class and the client. How do we know when to use which? Is there a reason we can't update the client to support some of this functionality?

Reply:
The client doesn't return content type (e.g: image/jp2 when retrieving frames) which should be verified by test case. I also think about updating client to support, but updating product code purely for testing is not a  good practice, besides I see existing tests ([link](https://github.com/microsoft/dicom-server/blob/882ff5f461b6946ea1d74797753245deeb8de86a/test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/RetrieveTransactionResourceTests.cs#L133)) already build httprequest by itself for validation, so I think this might be a better way.

I'm open to question, we can discuss to get the best solution.

Reply:
Positive test cases have been updated to use client. Still need this for Negative tests.